plugins/modules/helm_plugin_info.py

```diff
@@ -0,0 +1,175 @@
+stderr:
+  type: str
+  description: Full `helm` command stderr, in case you want to display it or examine the event log
+  returned: always
+  sample: ''
+command:
+  type: str
+  description: Full `helm` command built by this module, in case you want to re-run the command outside the module or debug a problem.
+  returned: always
+  sample: helm plugin list ...
+plugin_list:
+  type: list
+  description: Helm plugin dict inside a list
+  returned: always
+  sample: {
+      "name": "env",
+      "version": "0.1.0",
+      "description": "Print out the helm environment."
+  }
+rc:
+  type: int
+  description: Helm plugin command return code
+  returned: always
+  sample: 1
+'''
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            binary_path=dict(type='path'),
+            release_namespace=dict(type='str', required=True, aliases=['namespace']),
+            plugin_name=dict(type='str',),
+            # Helm options
+            context=dict(type='str', aliases=['kube_context'], fallback=(env_fallback, ['K8S_AUTH_CONTEXT'])),
+            kubeconfig=dict(type='path', aliases=['kubeconfig_path'], fallback=(env_fallback, ['K8S_AUTH_KUBECONFIG'])),
+        ),
+        supports_check_mode=True,
+    )
+
+    bin_path = module.params.get('binary_path')
+    release_namespace = module.params.get('release_namespace')
+
+    # Helm options
+    kube_context = module.params.get('context')
+    kubeconfig_path = module.params.get('kubeconfig')
+
+    if bin_path is not None:
+        helm_cmd_common = bin_path
+    else:
+        helm_cmd_common = 'helm'
+
+    helm_cmd_common = module.get_bin_path(helm_cmd_common, required=True)
+
+    helm_cmd_common += " plugin"
+
+    if kube_context is not None:
+        helm_cmd_common += " --kube-context " + kube_context
+
+    if kubeconfig_path is not None:
+        helm_cmd_common += " --kubeconfig " + kubeconfig_path
+
+    helm_cmd_common += " --namespace=" + release_namespace
+
+    plugin_name = module.params.get('plugin_name')
+    helm_plugin_list = helm_cmd_common + " list"
+    rc, out, err = module.run_command(helm_plugin_list)
+    if rc != 0 or (out == '' and err == ''):
+        module.fail_json(
+            msg="Failed to get Helm plugin info",
+            command=helm_plugin_list,
+            stdout=out,
+            stderr=err,
+            rc=rc,
+        )
+
+    plugin_list = []
+    if out:
+        for line in out.splitlines():
+            if line.startswith("NAME"):
+                continue
+            name, version, description = line.split('\t', 3)
+            if plugin_name is not None and plugin_name == name:
+                plugin_list.append({
+                    'name': name.strip(),
+                    'version': version.strip(),
+                    'description': description.strip(),
+                })
+                break
+            plugin_list.append({
```

Comment:
Should we not do this when `plugin_name` has been set? I think currently the module will return info on all the plugins up to the one specified when `plugin_name` is set